lightningd/invoice.c

```diff
@@ -381,15 +272,12 @@ static void json_delinvoice(struct command *cmd,
 	 * otherwise the invoice will be freed. */
 	json_add_invoice(response, i);
 
-	error = delete_invoice(cmd, cmd->ld->wallet, invs, i);
+	error = wallet_invoice_delete(wallet, i);
 
 	if (error) {
 		log_broken(cmd->ld->log, "Error attempting to remove invoice %"PRIu64,
 			   i->id);
-		command_fail(cmd, "%s", error);
-		/* Both error and response are sub-objects of cmd,
-		 * and command_fail will free cmd (and also error
-		 * and response). */
+		command_fail(cmd, "Database error");
```

Comment:
A bit generic, this can only happen if the invoice cannot be found in the DB in the first place.

Reply:
Mostly copied from old code.  This may be more useful in the future when we implement deletions based on state as per #477 (in which case the error would be "Invoice not in specified state").